src/param.rs

```diff
@@ -300,6 +344,46 @@ impl AudioParam {
             value: 0., // no value
             time: cancel_time,
             time_constant: None,
+            cancel_time: None,
+            duration: None,
+            values: None,
+        };
+
+        self.send_event(event);
+    }
+
+    pub fn cancel_and_hold_at_time(&self, cancel_time: f64) {
+        let event = AudioParamEvent {
+            event_type: AudioParamEventType::CancelAndHoldAtTime,
+            value: 0., // value will be defined by cancel event
+            time: cancel_time,
+            time_constant: None,
+            cancel_time: None,
+            duration: None,
+            values: None,
+        };
+
+        self.send_event(event);
+    }
+
+    pub fn set_value_curve_at_time(&self, values: &[f32], start_time: f64, duration: f64) {
```

Comment:
If we need ownership of values, let's just require a `Vec<f32>`? The spec says we should make a copy indeed but that's because JS lacks ownership semantics

Reply:
I guess I see what you mean, but I'm not completely sure that's what we want here

For example,  in JS that would be very simple to end up writing something like that (e.g. granular synthesis w/ fancy dynamic window):

```js
const someCurve = new Float32Array(/*...*/); // some windowing function

someScheduler.tick(time => {
  const env = audioContext.createGain();
  env.setValueCurveAtTime(someCurve, time, duration);
  // or even (e.g. kind of `slice`?)
  env.setValueCurveAtTime(someCurve.subarray(start, end), time, duration);

  // @note - also, as `setValueCurveAtTime` makes an internal copy of the given values, `someCurve` 
  // could be mutated (e.g. scaling, widening) at any point without affecting already scheduled automations
}, period);
```

If `set_value_curve_at_time` takes ownership of `values`, such pattern wouldn't be possible anymore and the user would have to re-create the window each time, or am I missing something?

Reply:
Well someone has to clone the values, either the lib or the user. By putting the burden on the user, it will be possible to avoid useless clones (e.g. when the user is not interested in re-using the values). In any case, the current implementation is fine. Maybe something to add to the discussion in #20